messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/deployment/injection/InjectedJMSContext.java

```diff
@@ -141,4 +158,27 @@ private Object lookup(String name) {
             }
         }
     }
-}
+
+    /**
+     * Synchronization task, which executes "cleanup" on JMSContext in AfterCompletion phase. BeforeCompletion does nothing.
+     */
+    private class AfterCompletionSynchronization implements Synchronization {
```

Comment:
Maybe this class should be placed somewhere in TransactedJMSContext (because cleanup method for request JMSContexts is placed in  RequestedJMSContext). Transactional JMSContexts could behave in the same way.  @jmesnil What do you think?

Reply:
sounds good to move AfterCompletionSynchronization in TransactedJMSContext